### DISCUSSION TITLE
Is it possible to have multiple collections share the same env variables? I can't make sense of projects and workspaces

### DISCUSSION THREAD

**Discussion by victor-marino** [Kong/insomnia#6542] [Help] [answered]
Just wanted to double check if I'm missing something.

I have several environments set up for work: dev, staging, etc. Each of these environments has its own credentials and base URLs. Then I have to work with several APIs, which use paths relative to the base URLs of these environments.

In Postman, environments are defined at workspace level, which makes sense: that means I could import several collections (several APIs) inside the same workspace, and they would automatically share the same environment variables.

In Insomnia, however, it looks like environment variables are defined at _collection_ level? So if I import/create a new collection inside my current project, I would need to duplicate all the environments inside it.

The only other option I can see would be to add all my APIs as _folders_ inside the same collection... but then what's the point of having projects _and_ workspaces above collections?

Am I missing something? Is there no way to define variables at project or workspace level?

Also, I can't for the life of me figure out how to create a new workspace?

My understanding is that the hierarchy goes like this in Insomnia:
> Workspace > Project > Collection

On the left hand side, I have a single workspace called "Personal projects", and that's it. I can create new _projects_ inside this workspace (I created the "Work" one as you can see here), but it doesn't look like I have an option to create a new, separate workspace?

Screenshot of what I see: https://i.imgur.com/lhSc8pz.png

Can anyone help me clarify these points?

**Comment by juandevgado** [comment]
_Old question but in case someone lands from Google (like I did)_

Collections are isolated groups of requests, they are meant for use cases like microservices, legacy systems, etc where they share their own URLs, auth, patterns, etc.
If all you're querying is a single API just name your collection something like Main or API and use folders to group requests. Those will share env vars without issue.

**Reply by ItsCurstin** [reply]
thanks, I think this makes it easier especially for when using the cookies across different requests in the same collection. Since I noticed the cookies are not shared across different collections which is fine if that was the design choice

**Comment by subnetmarco** [comment]
Update: We will introduce this soon, most likely we will start work on this in May.

**Reply by TomDRichmond** [reply]
Any updated timeline on the introduction of cross-collection environments?

**Reply by subnetmarco** [reply]
About 2 weeks, it will ship in v9.3 GA.

**Comment by subnetmarco** [comment, marked as answer]
Global environments [has shipped](https://konghq.com/blog/product-releases/insomnia-9-3-ga) in v9.3 and you can now share environment variables across multiple files (like collections) in a project.
